components/merino/src/curation.rs

```diff
@@ -0,0 +1,200 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+use serde::{Deserialize, Serialize};
+
+// Locales supported by Merino Curated Reccomendations
+#[derive(Debug, Serialize, Deserialize, PartialEq, uniffi::Enum)]
```

Comment:
Nit: I assume `Locale` would only be used by consumers to specify the `locale` field in the POST request, then we can just derive `Serialize` for it w/o deriving `Deserialize` to reduce the compiling overhead as the compiler won't generate code for deserialization. 

In general, I think for structs or enums defined here, we can just pick either `Serialize` or `Deserialize` to derive unless it's used for both in some rare cases. Deriving `Deserialize` for structs/enums so we can load Merino API JSON response into Rust structs. Likewise, `Serialize` can be derived to serialize Rust structs into API requests.